web/src/main/java/org/springframework/security/web/util/matcher/DispatcherTypeRequestMatcher.java

```diff
@@ -66,7 +66,7 @@ public DispatcherTypeRequestMatcher(DispatcherType dispatcherType, @Nullable Htt
 	@Override
 	public boolean matches(HttpServletRequest request) {
 		if (this.httpMethod != null && StringUtils.hasText(request.getMethod())
-				&& this.httpMethod != HttpMethod.resolve(request.getMethod())) {
+				&& this.httpMethod != HttpMethod.valueOf(request.getMethod())) {
```

Comment:
as I read the deprecated method:

https://github.com/spring-projects/spring-framework/blob/6.0.x/spring-web/src/main/java/org/springframework/http/HttpMethod.java#L138

it seems to have a null-check. 

I wonder if you also need a null check here (`request.getMethod()` == null)?

Reply:
It's not needed since that condition is checked on line 68 with `StringUtils.hasText(request.getMethod())`